### PR TITLE
✨ Isolate test files

### PR DESCRIPTION
**Pull request by mboukhalfa**
isolating all test files and related varaibles under **tests** folder and keep vm_setup only for create the environment

### CODE REVIEW

**Comment by mboukhalfa** [comment]
keep-test-v1b1-ubuntu-e2e

**Comment by kashifest** [comment]
/hold
I am unsure about the necessity of this change. Please add some more detailed description to tell us about the idea behind this change.

**Comment by mboukhalfa** [comment]
> /hold I am unsure about the necessity of this change. Please add some more detailed description to tell us about the idea behind this change.

The current ansible tests are widespread in multiple folders some are under scripts other under vm-setup.
dev-env is not only about tests but it is 3 layer :
1- simulating the baremetal (installing virsh kvm and creating domains and network ..)
2- environment setup (spin up  minikube and run capi, capm3, bmo and ironic and having some BMH available)
3- tests (integration tests, feature tests that should be isolated in a folder an trivial use case for that if one day we deicided to remove ansible tests and keep only e2e it will be easier if ansible tests are isolated from other files)

moreover it will simplify the dev-env understanding with this modular structure then mixed files.

File: tests/feature_tests/remediation/remediation.sh [hunk 1]
```diff
@@ -5,8 +5,8 @@ METAL3_DIR="$(dirname "$(readlink -f "${0}")")/../../.."
 # shellcheck disable=SC1091
 # shellcheck disable=SC1090
 # shellcheck disable=SC2046
-source "${METAL3_DIR}/scripts/feature_tests/feature_test_vars.sh"
+source "../feature_test_vars.sh"
```
Comment (lentzi90) [inline tests/feature_tests/remediation/remediation.sh:8]: Please keep it as absolute path. Then we don't have to worry if someone runs it from an unexpected folder.

**Review by lentzi90 — COMMENTED** [review]
Overall good I think. The structure becomes clearer and easier to understand.

If we are going to do this I would also like to rename the `v1aX_integration_test` folder. It is not only for v1aX and it is not just integration tests, so what about `run_tests` or `tests` instead?

**Comment by kashifest** [comment]
> > /hold I am unsure about the necessity of this change. Please add some more detailed description to tell us about the idea behind this change.
> 
> The current ansible tests are widespread in multiple folders some are under scripts other under vm-setup. dev-env is not only about tests but it is mainly 3 layers : 1- **Simulating the baremetal** (installing virsh kvm and creating domains and network ..) 2- **Environment setup** (spin up minikube and run capi, capm3, bmo and ironic and having some BMH available) 3- **Tests** (integration tests, feature tests that should be isolated in a folder an trivial use case for that if one day we deicided to remove ansible tests and keep only e2e it will be easier if ansible tests are isolated from other files)
> 
> moreover it will simplify the dev-env understanding with this modular structure then mixed files.

Please add this in the description of the PR also.

One small concern, why are you leaving this [file](https://github.com/Nordix/metal3-dev-env/blob/isolate_test_files/mohammed/scripts/deploy_tilt_env.sh) behind? Thats also part of test right or is it because it has deploy , you are leaving it ? I am just thinking perhaps if we leave that behind, we rename the dir to `tilt` instead.

**Comment by mboukhalfa** [comment]
> > > /hold I am unsure about the necessity of this change. Please add some more detailed description to tell us about the idea behind this change.
> > 
> > 
> > The current ansible tests are widespread in multiple folders some are under scripts other under vm-setup. dev-env is not only about tests but it is mainly 3 layers : 1- **Simulating the baremetal** (installing virsh kvm and creating domains and network ..) 2- **Environment setup** (spin up minikube and run capi, capm3, bmo and ironic and having some BMH available) 3- **Tests** (integration tests, feature tests that should be isolated in a folder an trivial use case for that if one day we deicided to remove ansible tests and keep only e2e it will be easier if ansible tests are isolated from other files)
> > moreover it will simplify the dev-env understanding with this modular structure then mixed files.
> 
> Please add this in the description of the PR also.
> 
> One small concern, why are you leaving this [file](https://github.com/Nordix/metal3-dev-env/blob/isolate_test_files/mohammed/scripts/deploy_tilt_env.sh) behind? Thats also part of test right or is it because it has deploy , you are leaving it ? I am just thinking perhaps if we leave that behind, we rename the dir to `tilt` instead.

I thought this is not a test it just deploy environment it is part of environment setup, isn't it?

**Comment by furkatgofurov7** [comment]
> I thought this is not a test it is just deploying an environment with tilt! It is part of environment setup, isn't it?

It is used when we deploy dev-env and use "Tilt" as an `EPHEMERAL_CLUSTER`, I would say it is safe to consider that script as part of the new `tests` folder

**Comment by mboukhalfa** [comment]
@kashifest @furkatgofurov7 I think this file getting source in script 03 which get run by `make ` and not `make test` my understanding is `make` spin up the environment and `make test` or `make feature_tests` run the test so I still think that tilt script should not be part of the tests

**Comment by mboukhalfa** [comment]
@kashifest @furkatgofurov7 I think this file getting source in script 03 which get run by `make ` and not `make test` my understanding is `make` spin up the environment and `make test` or `make feature_tests` run the test so I still think that tilt script should not be part of the tests

**Comment by kashifest** [comment]
> @kashifest @furkatgofurov7 I think this tilt file getting source in script 03 which get run by `make ` and not `make test` my understanding is `make` spins up the environment and `make test` or `make feature_tests` run the test so I still think that tilt script should not be part of the tests

If you are reluctant to keep it there, please change the name to tilt-setup , I dont like `scripts` in the root dir and another `scripts`  in the test dir

**Comment by mboukhalfa** [comment]
> > It is not only for v1aX and it is not just integration tests
> 
> +1

done

**Comment by kashifest** [comment]
/approve
/hold cancel

Thanks @mboukhalfa

**Review by lentzi90 — COMMENTED** [review]
/lgtm

**Pull request merged**